servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -629,12 +633,15 @@ private void cancelAndTerminate0(Throwable cause) {
             private void scheduleQuery0(final long nanos) {
                 assertInEventloop();
 
-                LOGGER.trace("DnsClient {}, scheduling DNS query for {} after {} nanos.",
-                        DefaultDnsClient.this, AbstractDnsPublisher.this, nanos);
+                final long delay = ThreadLocalRandom.current()
+                        .nextLong(nanos, addWithOverflowProtection(nanos, scheduleJitterNanos));
+                LOGGER.debug("DnsClient {}, scheduling DNS query for {} after {}ms, original TTL: {}ms.",
+                        DefaultDnsClient.this, AbstractDnsPublisher.this, NANOSECONDS.toMillis(delay),
+                        NANOSECONDS.toMillis(nanos));
 
                 // This value is coming from DNS TTL for which the unit is seconds and the minimum value we accept
                 // in the builder is 1 second.
-                cancellableForQuery = nettyIoExecutor.schedule(this::doQuery0, nanos, NANOSECONDS);
+                cancellableForQuery = nettyIoExecutor.schedule(this::doQuery0, delay, NANOSECONDS);
```

Comment:
It would have been nice to support `retry strategies` here too, but can't think of an easy way to do so.

Reply:
SD retries are handled at client level, bcz it needs visibility into SD errors, see `SingleAddressHttpClientBuilder#retryServiceDiscoveryErrors(...)`.
Don't think we need retry `schedule`, `nettyIoExecutor` won't reject unless it's closed.